### PR TITLE
Passes status of scenario and failed reason

### PR DESCRIPTION
**Pull request by AFASbart**
For the flutter_gherkin package I have added the possibility to pass along the scenario's final status and to retry a specific step multiple times.

### CODE REVIEW

**Comment by AFASbart** [comment]
@jonsamwell Do you have the time to check this PR so we can resolve 3 (!) issues on the flutter_gherkin branch?

File: lib/src/configuration.dart [hunk 1]
```diff
@@ -85,6 +85,8 @@ class TestConfiguration {
     this.createWorld,
   });
 
+  int stepMaxRetries = 3;
```
Comment (jonsamwell) [inline lib/src/configuration.dart:88]: We should add documentation to this parameter.  Just something short.  I actually think this should default to zero to preserve existing behaviour.  It is actually bad practice to retry.  With cucumberjs to have to be explicit about the number of retries for a step / scenario

File: lib/src/gherkin/steps/step_definition.dart [hunk 2]
```diff
@@ -60,19 +60,16 @@ abstract class StepDefinitionGeneric<TWorld extends World> {
...
       );
-    } on Exception catch (e, st) {
-      return ErroredStepResult(
-        elapsedMilliseconds,
-        StepExecutionResult.error,
-        e,
-        st,
-      );
+    } on Error catch (e, st) {
+      return ErroredStepResult(elapsedMilliseconds, StepExecutionResult.error,
```
Comment (jonsamwell) [inline lib/src/gherkin/steps/step_definition.dart:64]: Add trailing comma and new line

**Review by jonsamwell — CHANGES_REQUESTED** [review]

**Comment by AFASbart** [comment]
Before merging, can you recheck the flutter PR: https://github.com/jonsamwell/flutter_gherkin/pull/221. The max retries need to be atleast 1 ofcourse (so I added a + 1 on the for loop) and I have added a delay after each failed attempt.

**Review by jonsamwell — APPROVED** [review]

**Pull request merged**